meta/templates/sai.thrift.tt

```diff
@@ -0,0 +1,248 @@
+
+[%- BLOCK define_attribute_list -%]
+// common attribute list
+struct sai_thrift_attribute_list_t {
+    1: list<sai_thrift_attribute_t> attr_list;
+    2: sai_thrift_int32_t attr_count;
+}
+[% END -%]
+
+[%- ######################################################################## -%]
+
+[%- ######################################################################## -%]
+
+[%- BLOCK function_debug_info -%]
+    [%- IF dbg -%]
+
+    // [% function.dbg_info %]
+
+        [%- IF function.object -%]
+            [%- IF function.operation == 'create' -%]
+    // [% function.object %] mandatory attrs: [% FOREACH attr IN apis.$api.objects.${function.object}.attrs.mandatory %][% attr.simple_name %], [% END %]
+
+            [%- END -%]
+            [%- IF function.operation == 'stats' -%]
+    // [% function.object %] stats: [% FOREACH stat IN apis.$api.objects.${function.object}.stats.all %][% stat.simple_name %], [% END %]
+            [%- ELSE -%]
+    // [% function.object %] mandatory attrs: [% FOREACH attr IN apis.$api.objects.${function.object}.attrs.${function.operation} %][% attr.simple_name %], [% END %]
+            [%- END %]
+
+        [%- END -%]
+    [%- END -%]
+[% END -%]
+
+[%- ######################################################################## -%]
+
+[%- ######################################################################## -%]
+
+[%- BLOCK function_declaration -%]
+    [% function.rpc_return.type.thrift_name %] [% function.thrift_name %](
+    [%- id = 1; comma = 0; FOREACH rpcarg IN function.args %]
+        [%- UNLESS rpcarg.internal %]
+            [%- IF comma %], [%- ELSE; comma = 1; END %]
+            [%- id; id = id + 1 %]: [% rpcarg.type.thrift_name %] [% rpcarg.name %]
+        [%- END %]
+    [%- END %]) throws (1: sai_thrift_exception e);
+[% END -%]
+
+[%- ######################################################################## -%]
+
+[%- ######################################################################## -%]
+
+[%- BLOCK define_api_functions -%]
+    [%- FOREACH function IN apis.$api.functions -%]
+        [%- PROCESS function_debug_info -%]
+
+        [%- PROCESS function_declaration -%]
+    [%- END -%]
+[% END -%]
+
+[%- ######################################################################## -%]
+
+[%- ######################################################################## -%]
+
+[%- BLOCK define_functions -%]
+    [%- IF apis.common.functions.size -%]
+
+    // common functions
+    [%- PROCESS define_api_functions api = 'common' -%]
+    [%- END -%]
+
+    [%- FOREACH api IN apis.keys.sort -%]
+        [%- IF apis.$api.functions.size and api != 'common' %]
+    // [% api %] API
+
+            [%- PROCESS define_api_functions -%]
+        [%- END -%]
+
+    [%- END -%]
+[% END -%]
+
+[%- ######################################################################## -%]
+
+[%- # The body of the file: -%]
+// AUTOGENERATED FILE! DO NOT EDIT
+
+namespace py sai
```

Comment:
current namespace is switch_sai. 
saiservee.cpp depends on those generated files, like switch_sai_rpc, do we plan to use a new saiserver.cpp?
saiserver is referenced by syncd-rpc, then we need to upgrade the syncd-rpc to use the new saiserver if we want to use the new structure?

Reply:
Yes. The 2nd PR for SAI PTF has a new sai_rpc_frontend.cpp which uses the new namespace. I am going to create a 3rd PR which will include the Makefile.